test/benchmark/tests/utils/s3.py

```diff
@@ -0,0 +1,96 @@
+import os
+import time
+import re
+from inspect import signature
+import boto3
+
+from retrying import retry
+from fabric2 import Connection
+from botocore.config import Config
+from botocore.exceptions import ClientError
+
+from invoke import run
+from invoke.context import Context
+
+from . import DEFAULT_REGION, IAM_INSTANCE_PROFILE, AMI_ID, LOGGER, S3_BUCKET_BENCHMARK_ARTIFACTS
+
+
+def upload_folder_to_s3(folder_path, s3_uri, connection=None):
+    """
+    Upload test-related artifacts to unique s3 location.
+    Allows each test to have a unique remote location for test scripts and files.
+    :param folder_path: path of the folder that must be uploaded
+    :param s3_uri: desired s3 path for the folder to reside in
+    :param connection: if provided, s3 command will be run on the ec2 instance
+    :return: True if uploaded succeeded else False
+    """
+
+    if connection:
+        run_out = connection.run(f"aws s3 cp --recursive {folder_path}/ {s3_uri}/")
+    else:
+        run_out = run(f"aws s3 cp --recursive {folder_path}/ {s3_uri}/")
```

Comment:
Using boto3 for both remote instance and local instance would be a little tedious. Since the client is already authenticated, this is simpler, and keeps the experience same when running locally, or on the ec2 instance. But I've noted this, and can be introduced in a later iteration for robustness.